### PR TITLE
Backfill course_id column of variants table with batched migration

### PR DESCRIPTION
**Pull request by SethPoulsen**
Use a batched migration to backfill the new course_id column of the variants table.

This is in preparation for Question Sharing, when the course_id of variant is no longer determined by the course_id of the question, and so the course_id must be saved in the variants table.

This PR must be merged sometime after:
- the course_id column is created in the variants table: #7273
- the batched migration infrastructure is set up: #7292

### CODE REVIEW

File: batched-migrations/20240210165255_variants_course_id.ts
Comment (nwalters512) [inline batched-migrations/20240210165255_variants_course_id.ts:1]: Despite the examples I show in the docs, PL does not yet support TypeScript :( we'll have to author these as a regular JavaScript file.

File: batched-migrations/20240210165255_variants_course_id.ts
Comment (nwalters512) [inline batched-migrations/20240210165255_variants_course_id.ts:17]:
```suggestion
        v.course_id = q.course_id
```

File: batched-migrations/20240210165255_variants_course_id.ts
Comment (SethPoulsen) [reply batched-migrations/20240210165255_variants_course_id.ts]: done

File: migrations/20240210165255_variants__course_id__backfill.js
Comment (nwalters512) [inline migrations/20240210165255_variants__course_id__backfill.js:1]: Seeing `2024` in the timestamp is a bit unexpected. Did you use the following command from https://github.com/PrairieLearn/PrairieLearn/blob/master/migrations/README.md#creating-a-migration?

```sh
node -e "console.log(new Date().toISOString().replace(/\D/g,'').slice(0,14))"
```

Comment (SethPoulsen) [reply migrations/20240210165255_variants__course_id__backfill.js:1]: LOL no. When I have a PR that is WIP I sometimes just put the timestamp way ahead so that I don't have to keep editing it when new PRs get merged. Then when its closer to ready I put an actual timestamp on. I'll change these once done debugging.

Comment (nwalters512) [reply migrations/20240210165255_variants__course_id__backfill.js:1]: Fair enough! FWIW, if you want to save yourself the trouble, we don't mandate that all new migrations have timestamps ahead of all the ones on master (that was a primary motivation for switching to timestamps from sequential integer indexes). As long as any migrations on which this one depends are ordered before it, you'll be good.

Comment (SethPoulsen) [reply migrations/20240210165255_variants__course_id__backfill.js:1]: Ah, that's cool. I must admit (as is now apparent I suppose) I didn't spend enough time looking at the new migration system to understand it.

**Comment by nwalters512** [comment]
Going to mark this as "do not merge". We need to ensure that #7273 has been fully deployed to all environments before landing this.

**Review by nwalters512 — APPROVED** [review]

**Pull request merged**
